acestep/ui/gradio/events/training/dataset_ops.py

```
@@ -13,6 +13,32 @@
+        base = ap.rsplit(".", 1)[0]
+        path = base + ".json"
+
+        # Remove None / empty keys so we don't overwrite existing metadata with garbage.
+        cleaned = {k: v for k, v in (meta or {}).items() if v is not None and v != ""}
+        import json
+        import os
+
+        os.makedirs(os.path.dirname(path), exist_ok=True)
+        with open(path, "w", encoding="utf-8") as f:
+            json.dump(cleaned, f, ensure_ascii=False, indent=2)
+        return path, ""
```

Ruff (0.15.0)

[warning] Consider moving this statement to an `else` block

(TRY300)